pkg/csi/service/wcp/controller.go

```diff
@@ -1085,6 +1076,17 @@ func (c *controller) ControllerUnpublishVolume(ctx context.Context, req *csi.Con
 					})
 					if err != nil {
 						if isStillAttached {
+							nodeName, err := commonco.ContainerOrchestratorUtility.GetNodeForAttachedVolume(ctx, req.VolumeId)
+							if err != nil {
+								log.Errorf("error while fetching the node name in pod %q spec for volumeId %q",
+									podVM.Reference().String(), req.VolumeId)
+							}
+							if nodeName != req.NodeId {
+								log.Infof("Node name in the pod spec is different. Assuming that the pod using the volume is "+
+									"scheduled on different node. Returning success for ControllerUnPublishVolume for "+
+									"volumeId: %q and VA: %q", req.VolumeId, volumeAttachment.Name)
+								break
+							}
 							log.Errorf("volume %q is still attached to node %q and podVM %q with moId %q", req.VolumeId,
 								req.NodeId, podVM.Reference().String(), podVM.Reference().Value)
 						}
```

Comment:
We want to return a non-storage fault if it's still attached and we fail the operation. Will that be part of another MR?

Reply:
Added